src/intptrcast.rs

```diff
@@ -0,0 +1,20 @@
+use std::cell::RefCell;
+
+use rustc::mir::interpret::AllocId;
+
+pub type MemoryState = RefCell<GlobalState>;
+
+#[derive(Clone, Debug)]
+pub struct GlobalState {
+    pub vec: Vec<(u64, AllocId)>,
```

Comment:
name this field `int_to_ptr_map` or sth (and throw some docs on it explaining the tuple)